server/jobs/storage_backend.go

```diff
@@ -20,86 +20,50 @@ const OutputPrefix = "output"
+	Read(ctx context.Context, key string) ([]string, error)
 
 	// Write logs to the storage backend
-	Write(key string, logs []string, fullRepoName string) (bool, error)
+	Write(ctx context.Context, key string, logs []string, fullRepoName string) (bool, error)
 }
 
-func NewStorageBackend(jobs valid.Jobs, logger logging.Logger, featureAllocator feature.Allocator, scope tally.Scope) (StorageBackend, error) {
-
-	if jobs.StorageBackend == nil {
-		return &NoopStorageBackend{}, nil
-	}
-
-	config := jobs.StorageBackend.BackendConfig.GetConfigMap()
-	backend := jobs.StorageBackend.BackendConfig.GetConfiguredBackend()
-	containerName := jobs.StorageBackend.BackendConfig.GetContainerName()
-
-	location, err := stow.Dial(backend, config)
-	if err != nil {
-		return nil, err
-	}
-
-	storageBackend := &storageBackend{
-		location:      location,
-		logger:        logger,
-		containerName: containerName,
-	}
-
+func NewStorageBackend(client *storage.Client, logger logging.Logger, featureAllocator feature.Allocator, scope tally.Scope) (StorageBackend, error) {
 	return &InstrumentedStorageBackend{
-		StorageBackend: storageBackend,
-		scope:          scope.SubScope("storage_backend"),
+		StorageBackend: &storageBackend{
+			client: client,
+			logger: logger,
+		},
+		scope: scope.SubScope("storage_backend"),
 	}, nil
 }
 
 type storageBackend struct {
-	location      stow.Location
-	logger        logging.Logger
-	containerName string
+	client *storage.Client
+	logger logging.Logger
 }
 
-func (s *storageBackend) Read(key string) (logs []string, err error) {
-	s.logger.Info(fmt.Sprintf("reading object for job: %s in container: %s", key, s.containerName))
-
-	container, err := s.location.Container(s.containerName)
+func (s *storageBackend) Read(ctx context.Context, key string) ([]string, error) {
+	s.logger.Info(fmt.Sprintf("reading object for job: %s", key))
+	reader, err := s.client.Get(ctx, key)
 	if err != nil {
-		return []string{}, errors.Wrap(err, "resolving container")
-	}
-
-	key = fmt.Sprintf("%s/%s", OutputPrefix, key)
-	item, err := container.Item(key)
-	if err != nil {
-		return []string{}, errors.Wrap(err, "getting item")
-	}
-
-	r, err := item.Open()
-	if err != nil {
-		return []string{}, errors.Wrap(err, "reading item")
+		return nil, errors.Wrap(err, "getting item")
 	}
+	defer reader.Close()
 
 	buf := new(strings.Builder)
-	_, err = io.Copy(buf, r)
+	_, err = io.Copy(buf, reader)
 	if err != nil {
 		return []string{}, errors.Wrapf(err, "copying to buffer")
 	}
 
-	logs = strings.Split(buf.String(), "\n")
-	return
+	logs := strings.Split(buf.String(), "\n")
+	return logs, nil
 }
 
-func (s *storageBackend) Write(key string, logs []string, _ string) (bool, error) {
-	container, err := s.location.Container(s.containerName)
-	if err != nil {
-		return false, errors.Wrap(err, "resolving container")
-	}
-
+func (s *storageBackend) Write(ctx context.Context, key string, logs []string, _ string) (bool, error) {
```

Comment:
Why is there an empty arg? Is that still necessary?

Reply:
Ahh didn't catch that! I have removed it now